ocaml/testsuite/tests/typing-layouts/allow-illegal-crossing.ml

```diff
@@ -0,0 +1,654 @@
+(* TEST
+ {
+   flags = "-allow-illegal-crossing";
+   expect;
+ }
+*)
+
+(******************************************************************************)
+(* Test 1: types can cross the portability and contention axes when annotated *)
+
+type a : value mod nonportable
+type b : value mod portable = private a
```

Comment:
I think this should be rejected. If we aren't remembering the kinds of private abbreviations (and we're not), then we should check that the kind annotation is a superkind of the kind of the RHS, and it's not here. Ditto for other examples.

Given that we've now discovered that private abbreviations are not nominative types (and therefore not so interesting in the context of this patch), I'm going to skip the other private-abbreviation tests.